### ISSUE TITLE
Unable to Update Maven Project with Selenium Java 3.4.0, facing NullPointerException in Maven Build and Update

### ISSUE THREAD

**Issue by nylifeinnj** [SeleniumHQ/selenium#3895]
OS:  
Windows 10 Pro 64-bit 14.393
Selenium Version:  
Java 3.4.0
Browser:  
N/A
IDE:
Eclipse Neon .3

Since the release of Selenium Java 3.4.0 client, I have been trying to pull the Maven Dependency for Selenium Java 3.4.0:
<dependency>
	    <groupId>org.seleniumhq.selenium</groupId>
	    <artifactId>selenium-java</artifactId>
	    <version>3.4.0</version>
</dependency>

While I was able to create a clean Maven package, every time I attempted to update the project, I have received a NullPointException:

![image](https://cloud.githubusercontent.com/assets/23008759/25354851/9824cbf4-2902-11e7-8109-73a18a1763a2.png)

Since my IDE displays no other logs, I am unable to pinpoint the cause of this issue. I have tried deleting the metadata,, .settings, and .project folders/file and re imported the Maven project, but still faced the same issue. 

There were times that I have received a NullPointerException even while the Maven build was successful, but I am unable to find a root cause since there were no stack traces:

![image](https://cloud.githubusercontent.com/assets/23008759/25355116/8b88492e-2903-11e7-8e69-dbfefd53293b.png)

Please note that I have reverted to Selenium Java 3.3.1 and even 3.2.0, but I have not received this error when I tried to update the project.

**Comment by barancev** [comment]
It's Eclipse-Maven integration issue, we hardly can help here.

**Comment by zniff** [comment]
I doubt it is an eclipse problem - because as nylifeinnj  reports the ONLY selenium java version affected is 3.4.0.
I get the exactly same error when setting version 3.4.0 in the pom, but not when i revert to 3.3.1 or others.
And executing a Testcase with 3.4.0 is not possible as well within eclipse but works fine with all other versions.

![webdriver_pom xml-eclipse](https://cloud.githubusercontent.com/assets/7913513/25425250/977ff7ae-2a6b-11e7-929b-b8dd1d2fe53b.png)

Here the stacktrace from eclipse and it's version

```
java.lang.NullPointerException
	at java.util.zip.ZipFile.getZipEntry(Unknown Source)
	at java.util.zip.ZipFile.access$900(Unknown Source)
	at java.util.zip.ZipFile$ZipEntryIterator.next(Unknown Source)
	at java.util.zip.ZipFile$ZipEntryIterator.nextElement(Unknown Source)
	at java.util.zip.ZipFile$ZipEntryIterator.nextElement(Unknown Source)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getJavaDocPathInArchive(BuildPathManager.java:692)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getJavaDocUrl(BuildPathManager.java:677)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getJavaDocUrl(BuildPathManager.java:670)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.configureAttchedSourcesAndJavadoc(BuildPathManager.java:306)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getClasspath(BuildPathManager.java:255)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getClasspath(BuildPathManager.java:363)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getClasspath(BuildPathManager.java:343)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.getClasspath(BuildPathManager.java:371)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.updateClasspath(BuildPathManager.java:191)
	at org.eclipse.m2e.jdt.internal.BuildPathManager.mavenProjectChanged(BuildPathManager.java:180)
	at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryManager.notifyProjectChangeListeners(ProjectRegistryManager.java:783)
	at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryManager.applyMutableProjectRegistry(ProjectRegistryManager.java:930)
	at org.eclipse.m2e.core.internal.project.registry.ProjectRegistryManager.refresh(ProjectRegistryManager.java:300)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.getProjectFacade(MavenBuilder.java:154)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod$1.call(MavenBuilder.java:89)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:176)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.execute(MavenBuilder.java:86)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder.build(MavenBuilder.java:200)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

	
eclipse.buildId=4.6.3.M20170301-0400
java.version=1.8.0_131
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_CH
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product -clean

```

**Comment by thred** [comment]
It's the selenium-java-3.4.0-javadoc.jar file:

```
$ jar -tf selenium-java-3.4.0-javadoc.jar
java.lang.NullPointerException
        at java.util.zip.ZipFile.getZipEntry(ZipFile.java:566)
        at java.util.zip.ZipFile.access$900(ZipFile.java:60)
        at java.util.zip.ZipFile$ZipEntryIterator.next(ZipFile.java:524)
        at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:499)
        at java.util.zip.ZipFile$ZipEntryIterator.nextElement(ZipFile.java:480)
        at sun.tools.jar.Main.list(Main.java:1118)
        at sun.tools.jar.Main.run(Main.java:293)
        at sun.tools.jar.Main.main(Main.java:1288)
```
Delete the org/seleniumhq folder in your Maven repo and disable "Download Artifact JavaDoc" in Eclipse as a workaround.

**Comment by jansohn** [comment]
The problem is the empty file `selenium-java-3.4.0-javadoc` in the javadoc file. Java cannot handle such entries currently (fixed for JDK9): https://bugs.openjdk.java.net/browse/JDK-8048990

The empty file should be removed from the javadoc file.

**Comment by neo-de** [comment]
Wasted quite some time. Thanks @thred for the workaround.

**Comment by vinayakcc** [comment]
This was perplexing at best. Thanks for the workaround.
@thred, My environment eclipse already had "Download Artifact JavaDoc" option disabled and yet was facing issue.
This got resolved for me by explicitly deleting the selenium-java v3.4.0 javadoc related file(s) in local maven repository.

Cheers and happy coding!

**Comment by lourish** [comment]
Thanks for the workaround, this had me stumped for a fair while!.

In case you want to keep downloading javadocs and sources it also works if you copy javadocs from a previous version - I copied selenium-java-3.3.1-javadoc.jar, selenium-java-3.3.1-javadoc.jar.sha1 and renamed to selenium-java-3.4.0-javadoc.jar, selenium-java-3.4.0-javadoc.jar.sha1

**Comment by barancev** [comment]
Fixed in version 3.5.1
